### PR TITLE
[MSPAINT] Restore the main window saved show state

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Our `mspaint` didn't remember the maximized status of the main window.
JIRA issue: [CORE-18867](https://jira.reactos.org/browse/CORE-18867)

## Proposed changes

- Initialize `RegistrySettings::WindowPlacement` correctly.
- Fix `mainWindowPos` values.
- Add `nCmdShow` parameter to `RegistrySettings::Load` and `RegistrySettings::LoadPresets`.
- Use `registrySettings.WindowPlacement.showCmd` for `ShowWindow` call.

## TODO

- [x] Do tests.

### CODE REVIEW

File: base/applications/mspaint/main.cpp [hunk 1]
```diff
@@ -340,7 +340,7 @@ _tWinMain (HINSTANCE hThisInstance, HINSTANCE hPrevInstance, LPTSTR lpszArgument
     mainWindow.SetWindowPlacement(&(registrySettings.WindowPlacement));
 
     /* Make the window visible on the screen */
-    ShowWindow (hwnd, nFunsterStil);
```
Comment (HBelusca) [inline base/applications/mspaint/main.cpp:343]: You may replace as well "nFunsterStil" in the _tWinMain prototype above:
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=nFunsterStil
by `nShowCmd`  (standard name).

**Review by HBelusca — APPROVED** [review]
According to the fix, the actual title of this PR (+ associated commit) would be: "Restore the main window saved show state"

File: base/applications/mspaint/main.cpp (hunk 1 above)
Comment (katahiromz) [reply base/applications/mspaint/main.cpp:343]: Renamed as `nCmdShow`.

File: base/applications/mspaint/main.cpp [hunk 2]
```diff
@@ -337,10 +339,19 @@ _tWinMain (HINSTANCE hThisInstance, HINSTANCE hPrevInstance, LPTSTR lpszArgument
...
+    {
+        RECT rc = registrySettings.WindowPlacement.rcNormalPosition;
+        mainWindow.MoveWindow(rc.left, rc.top,
+                              rc.right - rc.left, rc.bottom - rc.top,
+                              TRUE);
+    }
 
     /* Make the window visible on the screen */
-    ShowWindow (hwnd, nFunsterStil);
+    if (registrySettings.WindowPlacement.showCmd == SW_MAXIMIZE)
```
Comment (HBelusca) [inline base/applications/mspaint/main.cpp:351]: I know for a fact, that this code (with all those ifs etc.) might be simplified.
For example:
- intializing to defaults the position: https://git.reactos.org/?p=reactos.git;a=blob;f=base/applications/mscutils/eventvwr/eventvwr.c;hb=de892d5bc715d68af757a36173e9f99e7927115f#l361
(and later, getting them from registry);
- next, creating the main window already at its correct position:
https://git.reactos.org/?p=reactos.git;a=blob;f=base/applications/mscutils/eventvwr/eventvwr.c;hb=de892d5bc715d68af757a36173e9f99e7927115f#l3041
- then just calling ShowWindow with the correct parameter:
https://git.reactos.org/?p=reactos.git;a=blob;f=base/applications/mscutils/eventvwr/eventvwr.c;hb=de892d5bc715d68af757a36173e9f99e7927115f#l3261

Comment (katahiromz) [reply base/applications/mspaint/main.cpp]: Fixed.

**Pull request merged**